src/benchmarklib/table_generator.cpp

```diff
@@ -108,8 +107,30 @@ std::shared_ptr<Table> TableGenerator::generate_table(
 
   pseudorandom_engine.seed(rd());
 
+  // Base allocators if we don't use multiple NUMA nodes
+  auto allocator_ptr_base_column = PolymorphicAllocator<std::shared_ptr<BaseColumn>>{};
+  auto allocator_value_column_int = PolymorphicAllocator<ValueColumn<int>>{};
+  auto allocator_chunk = PolymorphicAllocator<Chunk>{};
+  auto allocator_int = PolymorphicAllocator<int>{};
```

Comment:
For anyone who looks at this: I am not 100% sure if I am doing this correctly. Having to use the proper types `PolymorphicAllocator<BaseColumn/Chunk/int/etc>` in the specific cases (see lines 133, 180, 181, 186) makes sense here, right?

Reply:
Not sure how this would work. The data is in `value_vectors` and those vector just use the default allocator, and later you just move the data out of them into the ValueColumns and that should just take the pointer, not copy any data to nodes. 
Just a quick look from my side, maybe I miss something.

Reply:
That is correct. For ValueColumns, it is currently *not possible* to specify the Node they are placed on (see #402 for discussion). This will be changed once [this](https://github.com/hyrise/hyrise/issues/281#issuecomment-337953032) happens.

The placement on the desired Node happens in the DictionaryCompression, because for DictionaryColumns it *is possible*. Therefore we do need to set the correct Allocators in the ValueColumn, because although the ValueColumn does not use them itself, the DictionaryCompression fetches the Allocators from the ValueColumn for correct placement.

Reply:
This also answers your comment below, because yes, at this point (ValueColumn generation) we don't actually place the data on the correct nodes.

Reply:
Alright, gottcha. Pls add a comment explaining this somewhere in the method if its not explained already and I overlooked it.

Reply:
Yeah good point, I will add a comment in the header explaining that `use_multiple_numa_nodes` can only be used with compression because of bla bla

Reply:
About the original question: Yes and no. As far as I understand it, the template parameter is just so that the returned memory has the right type. The algorithm only cares for the size.

Reply:
Sounds about right. But in both cases, type and size, the allocator should match the type it is used for, or not?